controllers/gslb_controller_setup.go

```diff
@@ -0,0 +1,197 @@
+			err := c.List(context.TODO(), gslbList, opts...)
+			if err != nil {
+				log.Info().Msg("Can't fetch gslb objects")
+				return nil
+			}
+			gslbName := ""
+			for _, gslb := range gslbList.Items {
+				for _, rule := range gslb.Spec.Ingress.Rules {
+					for _, path := range rule.HTTP.Paths {
+						if path.Backend.Service != nil && path.Backend.Service.Name == a.GetName() {
+							gslbName = gslb.Name
+						}
+					}
+				}
+			}
+			if len(gslbName) > 0 {
+				return []reconcile.Request{
+					{NamespacedName: types.NamespacedName{
+						Name:      gslbName,
+						Namespace: a.GetNamespace(),
+					}},
+				}
+			}
+			return nil
+		})
+
+	ingressMapHandler := handler.EnqueueRequestsFromMapFunc(
+		func(a client.Object) []reconcile.Request {
+			annotations := a.GetAnnotations()
+			if annotationValue, found := annotations[strategyAnnotation]; found {
+				c := mgr.GetClient()
+				r.createGSLBFromIngress(c, a, annotationValue)
+			}
+			return nil
+		})
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&k8gbv1beta1.Gslb{}).
+		Owns(&netv1.Ingress{}).
+		Owns(&externaldns.DNSEndpoint{}).
+		Watches(&source.Kind{Type: &corev1.Endpoints{}}, endpointMapHandler).
+		Watches(&source.Kind{Type: &netv1.Ingress{}}, ingressMapHandler).
+		Complete(r)
+}
+
+func (r *GslbReconciler) createGSLBFromIngress(c client.Client, a client.Object, strategy string) {
+	log.Info().
+		Str("annotation", fmt.Sprintf("(%s:%s)", strategyAnnotation, strategy)).
+		Str("ingress", a.GetName()).
+		Msg("Detected strategy annotation on ingress")
+
+	ingressToReuse := &netv1.Ingress{}
+	err := c.Get(context.Background(), client.ObjectKey{
+		Namespace: a.GetNamespace(),
+		Name:      a.GetName(),
+	}, ingressToReuse)
+	if err != nil {
+		log.Info().
+			Str("ingress", a.GetName()).
+			Msg("Ingress does not exist anymore. Skipping Glsb creation...")
+		return
+	}
+	gslbExist := &k8gbv1beta1.Gslb{}
+	err = c.Get(context.Background(), client.ObjectKey{
+		Namespace: a.GetNamespace(),
+		Name:      a.GetName(),
+	}, gslbExist)
+	if err == nil {
+		log.Info().
+			Str("gslb", gslbExist.Name).
+			Msg("Gslb already exists. Skipping Gslb creation...")
+		return
+	}
+	gslb := &k8gbv1beta1.Gslb{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   a.GetNamespace(),
+			Name:        a.GetName(),
+			Annotations: a.GetAnnotations(),
+		},
+		Spec: k8gbv1beta1.GslbSpec{
+			Ingress: k8gbv1beta1.FromV1IngressSpec(ingressToReuse.Spec),
+		},
+	}
+
+	gslb.Spec.Strategy, err = r.parseStrategy(a.GetAnnotations(), strategy)
+	if err != nil {
+		log.Err(err).
+			Str("gslb", gslbExist.Name).
+			Msg("can't parse Gslb strategy")
+		return
+	}
+
+	err = controllerutil.SetControllerReference(ingressToReuse, gslb, r.Scheme)
+	if err != nil {
+		log.Err(err).
+			Str("ingress", ingressToReuse.Name).
+			Str("gslb", gslb.Name).
+			Msg("Cannot set the Ingress as the owner of the Gslb")
+	}
+
+	log.Info().
+		Str("gslb", gslb.Name).
+		Msg("Creating new Gslb out of Ingress annotation")
```

Comment:
nit: I know there was no change in this msg, but can we put here:
"Creating a new Gslb out of Ingress with 'k8gb.io/strategy' annotation"  (we don't care about ingresses that don't have this annotation)

(`k8gb.io/strategy` == `strategy`)